src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorEngine.java

```diff
@@ -0,0 +1,296 @@
+
+        final GenotypesContext genotypes = GenotypesContext.create(vc.getNSamples());
+        for ( final Genotype genotype : vc.getGenotypes() ) {
+            PerReadAlleleLikelihoodMap perReadAlleleLikelihoodMap = null;
+            if (stratifiedPerReadAlleleLikelihoodMap != null) {
+                perReadAlleleLikelihoodMap = stratifiedPerReadAlleleLikelihoodMap.get(genotype.getSampleName());
+            }
+
+            final GenotypeBuilder gb = new GenotypeBuilder(genotype);
+            for ( final GenotypeAnnotation annotation : genotypeAnnotations) {
+                if (addAnnot.test(annotation)) {
+                    annotation.annotate(ref, vc, genotype, gb, perReadAlleleLikelihoodMap);
+                }
+            }
+            genotypes.add(gb.make());
+        }
+
+        return genotypes;
+    }
+
+    private static final class AnnotationManager {
+
+        private final List<String> annotationGroupsToUse;
+        private final List<String> annotationsToUse;
+        private final List<String> annotationsToExclude;
+
+        private AnnotationManager(final List<String> annotationGroupsToUse, final List<String> annotationsToUse, final List<String> annotationsToExclude){
+            this.annotationGroupsToUse = annotationGroupsToUse;
+            this.annotationsToUse = annotationsToUse;
+            this.annotationsToExclude = annotationsToExclude;
+
+            final Set<String> allAnnotationNames = new HashSet<>(AnnotationManager.getAllAnnotationNames());
+            final Set<String> unknownAnnots = Sets.difference(new HashSet<>(annotationsToUse), allAnnotationNames);
+            if (!unknownAnnots.isEmpty()){
+                throw new BadArgumentValue("annotation", "Annotations " + unknownAnnots + " were not found; please check that you have specified the annotation name correctly");
+            }
+
+            final Set<String> unknownAnnotsExclude = Sets.difference(new HashSet<>(annotationsToExclude), allAnnotationNames);
+            if (!unknownAnnotsExclude.isEmpty()){
+                throw new BadArgumentValue("annotation", "Excluded annotations " + unknownAnnots + " were not found; please check that you have specified the annotation name correctly");
+            }
+
+            final Set<String> unknownGroups =  Sets.difference(new HashSet<>(annotationGroupsToUse), new HashSet<>(AnnotationManager.getAllAnnotationGroupNames()));
+            if (!unknownGroups.isEmpty()){
+                throw new BadArgumentValue("group", "Unknown annotation group in " + unknownGroups + ". Known groups are " + knownSubInterfaceSimpleNames(AnnotationKind.class));
+            }
+        }
+
+        private List<String> knownSubInterfaceSimpleNames(Class<AnnotationKind> annotationKindClass) {
+            return knownSubInterfaces(annotationKindClass).stream().map(c ->c.getSimpleName()).collect(Collectors.toList());
+        }
+
+        private Set<Class<?>> knownSubInterfaces(final Class<?> clazz) {
```

Comment:
And document the fact that it only finds subinterfaces in the same package as `clazz`